### PR TITLE
Add R search API example

### PR DESCRIPTION
**Pull request by lorenzwalthert**
## What changes are proposed in this pull request?

Add an example for the R language to the search html doc. There were Python and Java examples already.

## How is this patch tested?

inspecting the docs after rendering with 
``` 
cd docs
make livehtml
```

And executing the command in an existing project to make sure it succeeds.

As suggested by the contributing guide.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [x] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/search-syntax.rst [hunk 1]
```diff
@@ -185,6 +185,19 @@ To search all known experiments for any MLflow runs created using the Inception
...
+^^^^^^
+The R API is similar to the Python API.
+
+.. code-block:: r
+
+  library(mlflow)
+  mlflow_search_runs(
+    'tags.experiment_iteration = "1"',
+    experiment_ids = as.character(2:3),
+    order_by = "tags.start_time DESCR"
```
Comment (mparkhe) [inline docs/source/search-syntax.rst:198]: DESC

File: docs/source/search-syntax.rst [hunk 2]
```diff
@@ -185,6 +185,19 @@ To search all known experiments for any MLflow runs created using the Inception
...
 
+R
+^^^^^^
+The R API is similar to the Python API.
+
+.. code-block:: r
+
+  library(mlflow)
+  mlflow_search_runs(
+    'tags.experiment_iteration = "1"',
```
Comment (mparkhe) [inline docs/source/search-syntax.rst:196]: can you please add `filter=...` for filter argument here?

**Review by mparkhe — CHANGES_REQUESTED** [review]
Thank for the docs fix. 1 bug and 1 recommended change.

File: docs/source/search-syntax.rst (hunk 2 above)
Comment (lorenzwalthert) [reply docs/source/search-syntax.rst]: Sure.

File: docs/source/search-syntax.rst (hunk 1 above)
Comment (lorenzwalthert) [reply docs/source/search-syntax.rst]: oops. Sorting does not work with tags apparently. I used a parameter now in my updated example.

**Comment by lorenzwalthert** [comment]
I changed the example to:

- name the `filter` argument.
- use an `and` condition in the filter and `>` to demonstrate more capabilities of the search API.
- fix `DESC` keyword in `order_by` and make argument a parameter not tag (the latter does not seem to work with `order_by`).

**Comment by dbczumar** [comment]
@mparkhe Can you take another look at this?

File: docs/source/search-syntax.rst (hunk 1 above)
Comment (mparkhe) [reply docs/source/search-syntax.rst]: `filter` and `order_by` should work with tags columns. the issue here was `DESCR` instead  of `DESC`

**Review by mparkhe — APPROVED** [review]
Thanks for fixing the example.

**Pull request merged**